dev/effect.nix

```diff
@@ -5,15 +5,21 @@
       {
         onPush.default.outputs.effects = {
           darwin-deploy = hci-effects.runIf (pkgs.lib.hasPrefix "refs/heads/gh-readonly-queue/master/" config.repo.ref)
-            (hci-effects.runNixDarwin {
-              config = self.darwinConfigurations.darwin02;
+            (hci-effects.mkEffect {
+              # using the drv path here avoids downloading the closure on the deploying machine
+              drv = builtins.unsafeDiscardStringContext self.darwinConfigurations.darwin02.config.system.build.toplevel.drvPath;
               secretsMap.hercules-ssh = "hercules-ssh";
-              ssh.destination = "m1@darwin02.nix-community.org";
-              userSetupScript = ''
+              effectScript = ''
                 writeSSHKey hercules-ssh
                 cat >>~/.ssh/known_hosts <<EOF
                 darwin02.nix-community.org ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIBt6uTauhRbs5A6jwAT3p3i3P1keNC6RpaA1Na859BCa
                 EOF
+                ${hci-effects.ssh { destination = "m1@darwin02.nix-community.org"; } ''
+                  set -eu
+                  newProfile=$(nix-store --realise $drv)
+                  sudo -H nix-env --profile /nix/var/nix/profiles/system --set $newProfile
+                  $newProfile/sw/bin/darwin-rebuild activate
```

Comment:
Do we also need `${build}/activate-user`:

```
    nix-copy-closure --to $target --gzip --use-substitutes ${build}
    ssh $target "
      sudo nix-env -p /nix/var/nix/profiles/system --set ${build}/systemConfig
      ${build}/activate-user
      sudo ${build}/activate
    "
```

Reply:
`darwin-rebuild activate` will do both `${build}/activate-user` and `${build}/activate`.